proto/src/lib.rs

```diff
@@ -232,5 +232,50 @@ pub mod resource_recorder {
 }
 
 pub mod logger {
+    use std::str::FromStr;
+
+    use chrono::{DateTime, NaiveDateTime, Utc};
+    use shuttle_common::log::{LogItem as LogItemCommon, LogRecorder};
+
     include!("generated/logger.rs");
+
+    impl From<LogItem> for LogItemCommon {
+        fn from(value: LogItem) -> Self {
+            let LogLine {
+                service_name,
+                tx_timestamp,
+                data,
+            } = value.log_line.expect("log item to have log line");
+            let tx_timestamp = tx_timestamp.expect("log to have timestamp");
+            Self {
+                id: value.deployment_id.parse().unwrap_or_default(),
+                internal_origin: shuttle_common::log::Backend::from_str(&service_name)
+                    .expect("backend name to be valid"),
+                timestamp: DateTime::from_utc(
+                    NaiveDateTime::from_timestamp_opt(
+                        tx_timestamp.seconds,
+                        tx_timestamp.nanos.try_into().unwrap_or_default(),
+                    )
+                    .unwrap_or_default(),
+                    Utc,
+                ),
+                line: String::from_utf8(data).expect("line to be utf-8"),
+            }
+        }
+    }
+
+    impl LogRecorder
+        for logger_client::LoggerClient<
+            shuttle_common::claims::ClaimService<
+                shuttle_common::claims::InjectPropagation<tonic::transport::Channel>,
+            >,
+        >
+    {
+        fn record(&self, log: LogItemCommon) {
+            // TODO: Make async + error handling?
+            // self.send_logs(request)
+            //     .await
+            //     .expect("Failed to sens log line");
```

Comment:
Comments :smile:

Reply:
You might want to implement this for the new batcher instead. Then you don't have to worry about the `async` and `errors`.